### PR TITLE
JAMES-3361 JMAP Draft: sharee should not be able to modify mailbox rights

### PR DESCRIPTION
**Pull request by LanKhuat**

### CODE REVIEW

**Comment by chibenwa** [comment]
Can we have https://github.com/linagora/james-project/pull/3670/commits/7145bd3c16f0d6992d99c437adad41e92a114cd5 commit alone?

**Comment by chibenwa** [comment]
```
[a87b7541d443f66d677b4a0cd9c11dbbd03fe8cf] [ERROR]   MailboxManagerTest$RightTests.setRightsByPathShouldThrowWhenNotOwner:2834 
[a87b7541d443f66d677b4a0cd9c11dbbd03fe8cf] Expecting:
[a87b7541d443f66d677b4a0cd9c11dbbd03fe8cf]   <org.apache.james.mailbox.exception.MailboxNotFoundException: 12 can not be found>
[a87b7541d443f66d677b4a0cd9c11dbbd03fe8cf] to be an instance of:
[a87b7541d443f66d677b4a0cd9c11dbbd03fe8cf]   <org.apache.james.mailbox.exception.InsufficientRightsException>
[a87b7541d443f66d677b4a0cd9c11dbbd03fe8cf] but was:
[a87b7541d443f66d677b4a0cd9c11dbbd03fe8cf]   <"org.apache.james.mailbox.exception.MailboxNotFoundException: 12 can not be found
[a87b7541d443f66d677b4a0cd9c11dbbd03fe8cf] 	at org.apache.james.mailbox.store.StoreRightManager.assertHaveAccessTo(StoreRightManager.java:230)
[a87b7541d443f66d677b4a0cd9c11dbbd03fe8cf] 	at org.apache.james.mailbox.store.StoreRightManager.lambda$setRights$8(StoreRightManager.java:220)
[a87b7541d443f66d677b4a0cd9c11dbbd03fe8cf] 	at com.github.fge.lambdas.functions.FunctionChainer.lambda$sneakyThrow$49(FunctionChainer.java:74)
```

Related

**Comment by chibenwa** [comment]
We need to run rights tests if, and only if, the underlying mailbox manager supports rights.

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java [hunk 1]
```diff
@@ -148,14 +149,18 @@ public void applyRightsCommand(MailboxPath mailboxPath, ACLCommand mailboxACLCom
         assertSharesBelongsToUserDomain(mailboxPath.getUser(), mailboxACLCommand);
         MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
         block(mapper.findMailboxByPath(mailboxPath)
-            .flatMap(mailbox -> mapper.updateACL(mailbox, mailboxACLCommand)
+            .flatMap(Throwing.<Mailbox, Mono<Void>>function(mailbox -> {
+                assertHaveAccessTo(mailbox, session);
+
+                return mapper.updateACL(mailbox, mailboxACLCommand)
                 .flatMap(aclDiff -> eventBus.dispatch(EventFactory.aclUpdated()
```
Comment (Arsnael) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java:156]: indent

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
You can cherry-pick 2 commits here https://github.com/chibenwa/james-project/commits/jmap-draft-mailbox-right-fix

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
`[ERROR]   CassandraListingWithSharingTest.testListWithSharedMailboxUS » MailboxNotFound ...`

Sounds related

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by LanKhuat** [comment]
JIRA: https://issues.apache.org/jira/browse/JAMES-3361

**Pull request merged**